src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/client/prisonersearch/PrisonerSearchClient.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.hmppsalertsapi.client.prisonersearch
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.client.prisonersearch.dto.PrisonerDto
+
+@Component
+class PrisonerSearchClient(@Qualifier("prisonerSearchWebClient") private val webClient: WebClient) {
+
+  fun getPrisoner(prisonerId: String): PrisonerDto? {
+    return try {
+      webClient
+        .get()
+        .uri("/prisoner/{prisonerId}", prisonerId)
+        .retrieve()
+        .bodyToMono(PrisonerDto::class.java)
+        .block()
+    } catch (e: WebClientResponseException) {
```

Comment:
I'd catch WebClientResponseException.NotFound so that other status codes don't return null. I'd then add a second catch for other exceptions that rethrows a DownstreamServiceException. Those are translated into 502s

catch (e: Exception) {
        throw DownstreamServiceException("Get prisoner request failed", e)
      }